Model/ShimModel.php

```diff
@@ -23,6 +23,16 @@ class ShimModel extends Model {
 	 * @param string|null $ds
 	 */
 	public function __construct($id = false, $table = null, $ds = null) {
+		if ($warn = Configure::read('Shim.warnAboutRelationProperty')) {
+			if ($this->getAssociated()) {
+				$message = 'Relations must be defined using $this->initialized() in ' . get_class($this);
+				if (Configure::read('debug') && $warn === 'exception') {
+					throw new ShimException($message, 500);
+				}
+				trigger_error($message, E_USER_WARNING);
+			}
+		}
+
 		parent::__construct($id, $table, $ds);
```

Comment:
Shouldnt this come first for this to work?

Reply:
If parent::__construct is called first behaviours will be loaded. This means that if the behaviour uses $Model->bindModel an warning will be shown.Doing it like this it will only show errors for the public relations variables.